orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.groovy

```diff
@@ -217,129 +235,225 @@ class JedisExecutionRepository implements ExecutionRepository {
+     */
+    def currentPipelineIds = filteredPipelineIdsByJedis[jedisPool]
+    currentPipelineIds = currentPipelineIds.subList(0, Math.min(criteria.limit, currentPipelineIds.size()))
+
+    def currentObservable = retrieveObservable(
+      Pipeline,
+      executionsByPipelineKey(pipelineConfigId),
+      new Func1<String, Iterable<String>>() {
+        @Override
+        Iterable<String> call(String key) {
+          withJedis(jedisPool) { Jedis jedis ->
+            return criteria.statuses ? currentPipelineIds : jedis.zrevrange(key, 0, (criteria.limit - 1))
+          }
         }
-      }
-    }, queryByAppScheduler)
+      },
+      queryByAppScheduler,
+      jedisPool
+    )
+
+    if (jedisPoolPrevious.present) {
+      /**
+       * If configured, construct an observable the will retrieve pipelines from the secondary redis
+       */
+      def previousPipelineIds = filteredPipelineIdsByJedis[jedisPoolPrevious.get()]
+      previousPipelineIds = previousPipelineIds.subList(0, Math.min(criteria.limit, previousPipelineIds.size()))
+
+      def previousObservable = retrieveObservable(
+        Pipeline,
+        executionsByPipelineKey(pipelineConfigId),
+        new Func1<String, Iterable<String>>() {
+          @Override
+          Iterable<String> call(String key) {
+            withJedis(jedisPoolPrevious.get()) { Jedis jedis ->
+              return criteria.statuses ? previousPipelineIds : jedis.zrevrange(key, 0, (criteria.limit - 1))
+            }
+          }
+        },
+        queryByAppScheduler,
+        jedisPoolPrevious.get()
+      )
+
+      // merge primary + secondary observables
+      return Observable.merge(currentObservable, previousObservable)
+    }
+
+    return currentObservable
   }
 
   @Override
   Orchestration retrieveOrchestration(String id) {
-    withJedis { Jedis jedis ->
+    withJedis(getJedisPoolForId("orchestration:${id}")) { Jedis jedis ->
       retrieveInternal(jedis, Orchestration, id)
     }
   }
 
   @Override
   void deleteOrchestration(String id) {
-    withJedis { Jedis jedis ->
+    withJedis(getJedisPoolForId("orchestration:${id}")) { Jedis jedis ->
       deleteInternal(jedis, Orchestration, id)
     }
   }
 
   @Override
   Observable<Orchestration> retrieveOrchestrations() {
-    all(Orchestration)
+    def observables = [all(Orchestration, jedisPool)]
+    jedisPoolPrevious.map { jedisPool -> observables << all(Orchestration, jedisPool) }
+    return Observable.merge(observables)
   }
 
   @Override
   @CompileDynamic
   Observable<Orchestration> retrieveOrchestrationsForApplication(String application, ExecutionCriteria criteria) {
     def allOrchestrationsKey = appKey(Orchestration, application)
-    def filteredOrchestrationIds = null as List<String>
+
+    /**
+     * Fetch orchestration ids from the primary redis (and secondary if configured)
+     */
+    Map<Pool<Jedis>, List<String>> filteredOrchestrationIdsByJedis = [:].withDefault { [] }
```

Comment:
There's some overlap between `orchestrationsForApplication` and `pipelinesForPipelineConfigId` but it seemed more comprehensible to keep them separate (orchestrations are stored in an unsorted set vs pipelines, etc).